src/ckb_smt.rs

```diff
@@ -0,0 +1,109 @@
+use crate::collections::linked_list::LinkedList;
+use crate::H256;
+use core::{ffi::c_void, result::Result};
+
+#[link(name = "dl-c-impl", kind = "static")]
```

Comment:
choose another name instead of "dl-c-impl", e.g. "smt-c-impl" is OK.

Reply:
OK